### PR TITLE
BUG: Series[td64].sum() wrong on empty series, closes #37151

### LINKED ISSUE #37151: API/BUG: empty sum with td64 NaT or timedelta(0)?
```
ser = pd.Series([], dtype="m8[ns]")

>>> ser.sum()
NaT
>>> pd.Index(ser).sum()
NaT
>>> ser.array.sum()
NaT

>>> ser.to_frame().sum()
0   0 days
dtype: timedelta64[ns]
```

I think zero makes more sense.  Easy fix if we have consensus.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #37151
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Will handle min_count on the next pass.

### CODE REVIEW

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -381,14 +381,12 @@ def sum(
...
-        if not self.size and (self.ndim == 1 or axis is None):
-            return NaT
 
         result = nanops.nansum(
-            self._data, axis=axis, skipna=skipna, min_count=min_count
+            self._ndarray, axis=axis, skipna=skipna, min_count=min_count
         )
-        if is_scalar(result):
-            return Timedelta(result)
+        if axis is None or self.ndim == 1:
```
Comment (jreback) [inline pandas/core/arrays/timedeltas.py:388]: is the ndim >1 tested at all? why is this needed

File: pandas/tests/arrays/test_timedeltas.py [hunk 2]
```diff
@@ -187,6 +188,19 @@ def test_reductions_empty(self, name, skipna):
         result = getattr(arr, name)(skipna=skipna)
         assert result is pd.NaT
 
+    @pytest.mark.parametrize("skipna", [True, False])
+    def test_sum_empty(self, skipna):
+        tdi = pd.TimedeltaIndex([])
+        arr = tdi.array
+
+        result = tdi.sum(skipna=skipna)
```
Comment (jreback) [inline pandas/tests/arrays/test_timedeltas.py:196]: we should have a `min_count` argument if we are going to change this (I know its a PITA but if we do this then its yet another different as EA already has the min_count argument.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/test_timedeltas.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_timedeltas.py:196]: yah, planning on handling this in a dedicated pass.  these reductions are kind of a mess; ive already got four not-yet-pushed reduction branches on deck.

File: pandas/core/arrays/timedeltas.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py:388]: > is the ndim >1 tested at all?

yes, there are dedicated tests for this.

> why is this needed

1) completeness/correctness
2) fixing the reduction bugs is going to involve doing more things blockwise, not going through frame.values

**Review by jreback — APPROVED** [review]
